modules/dataframe/src/io/csvreader.cpp

```diff
@@ -608,11 +615,18 @@ std::shared_ptr<DataFrame> CSVReader::readData(std::istream& stream) const {
 
     if (firstRowHeader_) {
         rows.erase(rows.begin());
-        // remove lines skipped by filters
-        rows.erase(
-            std::remove_if(rows.begin(), rows.end(),
-                           [&](const auto& row) { return skipRow(row.first, row.second, false); }),
-            rows.end());
+
+        // demangle duplicate column headers similar to Pandas' default behavior
+        // @see https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html
+        std::unordered_map<std::string, int> headerCount;
+        for (auto& header : headers) {
+            if (int count = ++headerCount[header]; count > 1) {
+                header.append(fmt::format(".{}", count - 1));
```

Comment:
How will this interact with extraction of units?

Reply:
very good point.